### ISSUE TITLE
Tags parameter in taskcat.yml config file is not honoring for taskcat tests

### ISSUE THREAD

**Issue by UmaInformatica** [aws-quickstart/taskcat#449]
The issue has been reported to the AWS support team . They suggested us to report the issue in Github.

Issue reported :
We have a requirement to test our cloudformation in all the regions. For testing , we use taskcat tool.
Taskcat version installed was 0.9.8. The .taskcat.yml has below contents :
```
general:
  parameters:
    KeyName: 'keyname'
    InstanceType: 't2.small'
    SSHLocation: '0.0.0.0/32'
project:
  name:  test1040
  regions:
    - us-east-2
tests:
  dei:
    regions:
      - us-east-2
    tags:
      APPLICATIONENV: 'somevalue'
      APPLICATIONROLE: 'somevalue'
      BUSINESSUNIT: 'somevalue'
      NAME: 'somevalue'
      OWNEREMAIL: 'somevalue'
      RUNNINGSCHEDULE: 'somevalue'
    template: ./EC2InstanceWithSecurityGroupSample.template
```

We have specified tags in the config file but its not getting applying to cloudformation . Our company has strict policy on associating tags with every aws resource. Due to the above bug , we are unable to test our cloudformation solution using taskcat  .
Please let me know if you have any way to impose tags on the cloudformation.

Confirmation email from the AWS Support team :+1: 
From the correspondence, I understand you reached out to us regarding an issue you are currently facing with the taskcat tool, where the passed tags doesn't get applied to the CloudFormation stacks that are created using the tool.

To begin with, I tested this issue by running a sample project with a CloudFormation template using the taskcat tool, and I was successful in replicating your issue as the tags that were given in the taskcat config file didn't get applied to the CloudFormation stack. And the below given default tags were the only ones that were applied to the stack.

------------------------- CloudFormation Tags-------------------------------
Tags :

Key 					Value
taskcat-id		       aa3e32f7bed2427389de9ab9d987f3fb
taskcat-project-name	my-cfn-project-2
taskcat-test-name		my-test

-------------------------------------------------------------------------------------

Clearly, it looks like a bug that should be reported to the taskcat team, and since taskcat is an open source tool and is available to customers via github, I would suggest you to create an issue via the github page so that the developers working on this can address your issue directly, also this would help other people who are using the tool and encounter the same kind of problems. 

Also, I have escalated this issue internally so it can be looked over, unfortunately I don't have any ETA for this and any update should be directly communicated via the github issue.

_@andrew-glenn edit: Wrapping the provided config file in a code block_

**Comment by andrew-glenn** [comment]
I've been able to reproduce this. 

For background, it appears we're not passing tags to the `Stacker` class in the `test` CLI Module. 
https://github.com/aws-quickstart/taskcat/blob/84ecda3ed36288a125e9037d8e969f7a24e00954/taskcat/_cli_modules/test.py#L76-L81

The `Stacker` class accepts tags, but as a stand-alone list. We use this to pass taskcat metadata tags to the stack (`taskcat-id`, `taskcat-project-name`, `taskcat-test-name`). The `Stacker` class has logic to add these tags to each stack, but that's it. 

An action plan here. 
- Add a `tags` attribute to the `TestObj` class.
https://github.com/aws-quickstart/taskcat/blob/84ecda3ed36288a125e9037d8e969f7a24e00954/taskcat/_dataclasses.py#L301-L307

- Populate said attribute with the data from the `TestConfig` object. 
https://github.com/aws-quickstart/taskcat/blob/84ecda3ed36288a125e9037d8e969f7a24e00954/taskcat/_config.py#L316-L335

- Modify `Stacker` to look for tags within the test.
https://github.com/aws-quickstart/taskcat/blob/84ecda3ed36288a125e9037d8e969f7a24e00954/taskcat/_cfn/threaded.py#L51-L61

- Adjust unit tests.

**Comment by andrew-glenn** [comment]
This will be resolved when #450 is merged.
